src/dispatch/__init__.py

```diff
@@ -89,10 +92,35 @@ def run(init: Optional[Callable[P, None]] = None, *args: P.args, **kwargs: P.kwa
     parsed_url = urlsplit("//" + address)
     server_address = (parsed_url.hostname or "", parsed_url.port or 0)
     server = ThreadingHTTPServer(server_address, Dispatch(default_registry()))
+
+    for worker in _workers:
+        def entrypoint():
+            try:
+                worker()
+            finally:
+                server.shutdown()
+        _threads.append(threading.Thread(target=entrypoint))
+
+    for thread in _threads:
+        thread.start()
+
     try:
         if init is not None:
             init(*args, **kwargs)
         server.serve_forever()
     finally:
         server.shutdown()
         server.server_close()
+
+        for thread in _threads:
+            thread.join()
+
+def batch() -> Batch:
+    """Create a new batch object."""
+    return default_registry().batch()
+
+
+def worker(fn: Callable[None, None]) -> Callable[None, None]:
+    """Decorator declaring workers that will be started when dipatch.run is called."""
```

Comment:
Let's add more color about what a worker is to avoid confusion with other frameworks/workflow orchestrator systems which heavily use the same term.